index.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const cp = require('child_process');
+
+/**
+ * An error to be thrown when invokation of the cumulus-message-adapter fails
+ */
+class CumulusMessageAdapterExecutionError extends Error {
+  // eslint-disable-next-line require-jsdoc
+  constructor(message, fileName, lineNumber) {
+    super(message, fileName, lineNumber);
+    this.name = 'CumulusMessageAdapterExecutionError';
+  }
+}
+
+/**
+ * Invoke the cumulus-message-adapter
+ *
+ * @param {string} command - the action to be performed by the message-adapter
+ * @param {Object} input - the input to be sent to the message-adapter
+ * @returns {Promise.<Object>} - the output of the message-adapter
+ */
+function callCumulusMessageAdapter(command, input) {
+  return new Promise((resolve, reject) => {
+    const sled = cp.spawn('python', ['./cumulus-sled.zip', command]);
+
+    // Collect STDOUT
+    let sledStdout = '';
+    sled.stdout.on('data', (chunk) => {
+      sledStdout += chunk;
```

Comment:
Just curious - Why do we need the += here?

```javascript
let sledStdout;
sled.stdout.on('data', (data) => {
  sledStdout = data;
});
```
Is more in-line with the examples https://nodejs.org/api/child_process.html

Reply:
The 'data' event occur be called multiple times.